src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.ts

```diff
@@ -18,76 +18,140 @@
  */
 
 import angular from 'angular';
-import { Embeddable } from 'ui/embeddable';
-import searchTemplate from './search_template.html';
+import { SearchSource } from 'ui/courier';
 import * as columnActions from 'ui/doc_table/actions/columns';
-import { getTime } from 'ui/timefilter/get_time';
+import {
+  ContainerState,
+  Embeddable,
+  EmbeddableState,
+  OnEmbeddableStateChanged,
+  TimeRange,
+} from 'ui/embeddable';
+import { Filters, Query } from 'ui/embeddable/types';
 import { RequestAdapter } from 'ui/inspector/adapters';
+import { Adapters } from 'ui/inspector/types';
+import { getTime } from 'ui/timefilter/get_time';
+import { SavedSearch, SavedSearchLoader } from '../types';
+import searchTemplate from './search_template.html';
+
+interface SearchScope extends ng.IScope {
+  columns?: string[];
+  description?: string;
+  sort?: string[];
+  searchSource?: SearchSource;
+  sharedItemTitle?: string;
+  inspectorAdapters?: Adapters;
+  setSortOrder?: (column: string, columnDirection: string) => void;
+  removeColumn?: (column: string) => void;
+  addColumn?: (column: string) => void;
+  moveColumn?: (column: string, index: number) => void;
+  filter?: (field: string, value: string, operator: string) => void;
+}
+
+interface SearchEmbeddableCustomization {
+  sort?: string[];
+  columns?: string[];
+}
+
+interface SearchEmbeddableConfig {
+  onEmbeddableStateChanged: OnEmbeddableStateChanged;
+  savedSearch: SavedSearch;
+  editUrl: string;
+  loader: SavedSearchLoader;
```

Comment:
This should be removed, since we don't pass the loader in. It currently only works, because `search_embeddable_factory` is still in JS.